app/js/views/dev/camera-latest-image.jsx

```diff
@@ -0,0 +1,106 @@
+import React from 'components/react';
+
+import BaseView from 'js/views/base-view';
+
+export default class CameraLatestImage extends BaseView {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      service: null,
+      hasPreview: false
+    };
+    this.foxbox = props.foxbox;
+  }
+
+  componentDidMount() {
+    this.foxbox.getService(this.props.id).then((response) => {
+      const service = response.data;
+
+      this.setState({ service });
+
+      return this.foxbox.performGetOperation(
+        this.getGetOperation(service, 'latest image')
+      );
+    })
+    .then((image) => {
+      this.refs.snapshotPreview.src = URL.createObjectURL(image);
+      this.setState({ hasPreview: true });
+    })
+    .catch((e) => {
+      console.error(
+        'Error occurred while retrieving latest image for camera (id=%s): ',
+        this.props.id,
+        e
+      );
+    });
+  }
+
+  renderHeader() {
+    return super.renderHeader(
+      this.state.service && this.state.service.properties.name ?
+        this.state.service.properties.name :
+        'Unknown Service'
+    );
+  }
+
+  renderBody() {
+    let cameraControlsClass = 'app-view__fill-body camera-controls';
+    if (this.state.hasPreview) {
+      cameraControlsClass += ' camera-controls--has-preview';
+    }
+
+    return (
+      <div className={cameraControlsClass}>
+        <img ref="snapshotPreview"
+             style={{ flexGrow: 0 }}
```

Comment:
There no camera controls here, so it's the easiest fix to keep aspect ratio. Eventually we should just fix camera service view, I just don't want to spend time on this at the moment.